Docs: update commitment to OSS (#3465)
treeverse/lakeFS: docs/commitment.md

We believe in bottom up adoption of technologies.
We believe collaborative communities have the power to bring the best solutions to the community.
We believe that every engineer should be able to use, contribute and influence cutting edge technologies, so they can innovate in their domain.

***What is our commitment to open source?***

We created lakeFS, our open source project, to provide a Git-like interface on top of object stores - that you can fully take advantage of with any data application at any scale.

> In other words, we commit that the following capabilities are *and will remain* open source as part of lakeFS:

Reviewer: ```suggestion
For that reason, we commit that the following capabilities are *and will remain* open source as part of lakeFS:
```

  - All versioning capabilities
  - Git-Like interface for the versioning operations
  - Support for public object store APIs
  - Integrations with publicly available applications accessing an object store
  - CLI, API and GUI interfaces
  
We also commit to keeping lakeFS scalable in throughput and performance.

We are deeply committed to our [community of engineers](https://lakefs.io/community) who use and contribute to the project. We are and will continue to be highly responsive and shape lakeFS together to provide the data versioning capabilities we are all looking for.

***What is lakeFS Cloud?***

Treeverse offers lakeFS Cloud, which provides all the same benefits of the Git-like interface on top of object stores as a fully-managed service.

> lakeFS Cloud's vision is to provide a managed, usable data versioning and management solution for data practitioners, by leveraging the open source technology behind lakeFS and building added capabilities and unique features on top.

Reviewer: ```suggestion
The vision of lakeFS Cloud is to provide a managed data versioning and management solution for data practitioners. lakeFS Cloud  will leverage the lakeFS open source technology and add capabilities and unique features on top of it.
```

> As part of our commitment to the open source values of transparency and interoperability, we believe everyone should be able to enjoy these benefits, regarless of whether or not they choose to use the managed offering. 

Reviewer 2: small typo: regarless >> regardless
